### PR TITLE
feat: dev deploy remote bundles

### PR DESCRIPTION
**Pull request by decleaver**
## Description

allows you to run `uds dev deploy` with remote bundles

## Related Issue

Relates to #628 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [ ] Test, docs, adr added or updated as needed
- [ ] [Contributor Guide Steps](https://github.com/defenseunicorns/uds-cli/blob/main/CONTRIBUTING.md) followed

### CODE REVIEW

File: src/cmd/dev.go [hunk 1]
```diff
@@ -25,26 +24,30 @@ var devDeployCmd = &cobra.Command{
...
-		}
+		// Check if source is a local bundle
+		localBundle := helpers.IsDir(src)
+
+		if localBundle {
+			// Create Bundle
+			setBundleFile(args)
+
+			// Ensure source ends with a slash
+			if len(src) != 0 && src[len(src)-1] != '/' {
```
Comment (UncleGedd) [inline src/cmd/dev.go:44]: Why do we need the `/` at the end? (could also use `strings.HasSuffix` here)

File: README.md [hunk 2]
```diff
@@ -278,18 +278,19 @@ UDS CLI includes a vendored version of Zarf inside of its binary. To use Zarf, s
...
 
 ```
-uds dev deploy <path-to-bundle-yaml-dir>
+uds dev deploy <path-to-bundle-yaml-dir> | <oci-ref>
 ```
 
 The `dev deploy` command performs the following operations
-- Creates Zarf packages for all local packages in a bundle
+
+- (If local bundle) Creates Zarf packages for all local packages in a bundle
```
Comment (UncleGedd) [inline README.md:291]: Let's remove the parentheses from "(If local bundle)" and just use a colon, can also remove the "(If local bundle)" below and tab the "Creates a bundle from the..." over

File: src/pkg/sources/remote.go [hunk 3]
```diff
@@ -85,6 +90,16 @@ func (r *RemoteBundle) LoadPackage(dst *layout.PackagePaths, filter filters.Comp
 		}
 	}
 	addNamespaceOverrides(&pkg, r.nsOverrides)
+
+	if config.Dev {
```
Comment (UncleGedd) [inline src/pkg/sources/remote.go:94]: I think we have similar code in `sources/tarball.go` (same for the `ZarfInitConfig` check above), what do you think of refactoring these snippets to be in `sources/common.go`?

File: src/cmd/dev.go (hunk 1 above)
Comment (decleaver) [reply src/cmd/dev.go:44]: deleted.... was a band-aid for another bug. 🐛

File: src/pkg/sources/remote.go [hunk 4]
```diff
@@ -85,6 +90,11 @@ func (r *RemoteBundle) LoadPackage(dst *layout.PackagePaths, filter filters.Comp
...
+		YOLOPackage(&pkg)
```
Comment (UncleGedd) [inline src/pkg/sources/remote.go:95]: Naming suggestion: `setAsYOLO`

**Comment by UncleGedd** [comment]
Testing this locally, looks like I can't use `--set` to set a variable, let's add that flag (noting that setting vars through others means works fine)

**Pull request merged**